cmd/fyne_settings/settings/appearance.go

```diff
@@ -78,15 +109,45 @@ func (s *Settings) chooseTheme(name string) {
 	}
 	s.fyneSettings.ThemeName = name
 
-	switch name {
-	case "light":
-		s.preview.Resource = themeLightPreview
-	default:
-		s.preview.Resource = themeDarkPreview
-	}
+	s.preview.Image = s.createPreview()
 	canvas.Refresh(s.preview)
 }
 
+type overrideTheme interface {
+	OverrideTheme(fyne.Theme, string)
+}
+
+func (s *Settings) createPreview() image.Image {
+	c := playground.NewSoftwareCanvas()
+	oldTheme := fyne.CurrentApp().Settings().Theme()
+	oldColor := fyne.CurrentApp().Settings().PrimaryColor()
+
+	th := theme.DarkTheme()
+	if s.fyneSettings.ThemeName == "light" {
+		th = theme.LightTheme()
+	}
+	painter.SvgCacheReset() // reset icon cache
+	fyne.CurrentApp().Settings().(overrideTheme).OverrideTheme(th, s.fyneSettings.PrimaryColor)
+
+	empty := widget.NewLabel("")
+	tabs := widget.NewTabContainer(
+		widget.NewTabItemWithIcon("Welcome", theme.HomeIcon(), empty),
+		widget.NewTabItemWithIcon("Graphics", theme.DocumentCreateIcon(), empty),
+		widget.NewTabItemWithIcon("Widgets", theme.CheckButtonCheckedIcon(), makeProgressTab()),
+		widget.NewTabItemWithIcon("Containers", theme.ViewRestoreIcon(), empty),
+		widget.NewTabItemWithIcon("Windows", theme.ViewFullScreenIcon(), empty))
+	tabs.SetTabLocation(widget.TabLocationLeading)
+	tabs.SelectTabIndex(2)
+
+	c.SetContent(tabs)
+	c.Resize(fyne.NewSize(380, 380))
```

Comment:
I understand that this, and makeProgressTab below, are needed to showcase the chosen color, but I'm not a huge fan of the duplication, is there another way?

Reply:
I just copied and pasted this to show the feature - I am working on a replacement, it's this that is holding it in Draft state

Reply:
This is no longer in draft, but I don't see a replacement

Reply:
Erm, it's there: https://github.com/fyne-io/fyne/pull/1292/files#diff-fc4b8de1d5fbe30aec3d3d9d22e86237R119

Reply:
I don't follow.

> I'm not a huge fan of the duplication

> I am working on a replacement

_marked ready for review_

> I don't see a replacement

_link to the same duplicated code_

Reply:
Can you explain exactly what the duplication is? I think I am missing the point.
I coded a whole new tab container with different tabs, content and an overlay...
As per screenshot at the end of this conversation

Reply:
The duplication is this tabber, why not use fyne_demo tabber which already showcases various UI elements?

Reply:
because they are different apps and serve different purposes?
Fyne_demo is about to be completely rewritten using Tree anyhow so that concern is probably moot.

Reply:
Well the goal is to show how the primary colour looks across the UI, and so you'd probably want more than just a current tab and button (the only usage of primary colour in the screenshot below) which is why I was suggesting hooking into demo, but this is probably fine for a first attempt

Reply:
Yeah I played with progress bars but it looked awkward. Open to ideas of realistic content that could be dropped in though (under the overlay?)

Reply:
I had a look through fyne_demo to see what components we could put here, but there aren't many that show primary color when not in focus (which is good in general, but bad for this particular situation). I think what you have here so far is solid, we can always revisit in future.
What if the username entry in the login dialog could put into focus, so there is at least 3 uses of primary color?